### PR TITLE
Feature/support cosmos

### PR DESCRIPTION
**Pull request by hmoazam**
Implemented support for Azure Cosmos

### CODE REVIEW

**Comment by wjohnson** [comment]
@hmoazam would you please also add a unit test to the QNParsers file too?
https://github.com/hmoazam/Purview-ADB-Lineage-Solution-Accelerator/blob/feature/support-kusto/function-app/adb-to-purview/tests/unit-tests/Function.Domain/Helpers/Parser/QnParserTests.cs#L103-L107

```
[InlineData("OpenLineageNamespaceValue, 
                    "OpenLineageNameValue", 
                    "ExpectedFQN")]
```

File: function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs [hunk 1]
```diff
@@ -204,15 +242,28 @@ private async Task<bool> JoinEventData(Event olEvent, string jobRunId)
...
-                try
-                {
-                    var delresp = await _tableClient.DeleteEntityAsync(TABLE_PARTITION, olEvent.Run.RunId);
-                }
-                catch (Exception ex)
-                {
-                    _log.LogError(ex, $"OlMessageConsolodation-JoinEventData: Error {ex.Message} when deleting entity");
-                }
+            // Add Inputs to event if not already there (will only be done for DataSourceV2 sources)
+            if (olEvent.Inputs.Count == 0) {
```
Comment (wjohnson) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs:246]: Is there a case where there may be an input present initially and then more inputs later on?

What about a scenario where ABFSS is used and joined with a Data Source V2 source like Cosmos?

File: function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs [hunk 2]
```diff
@@ -228,11 +279,32 @@ private bool IsStartEventEnvironment(Event olEvent)
             return false;
         }
 
+        /// <summary>
+        /// Helper function to determine if the event is one of
+        /// the data source v2 ones which need to aggregate data
+        /// from the start and complete events
+        /// </summary>
+        private bool isDataSourceV2Event(Event olEvent) {
+            string[] special_cases = {"azurecosmos://", "iceberg://"}; // todo: make this configurable?
```
Comment (wjohnson) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs:288]: Agreed, we should definitely make this part of the configuration instead.  Maybe something like openLineageDataSourceV2Prefixes as the parameter and have it separated by semicolon (;)

Similar to the [Spark_Entities config](https://github.com/microsoft/Purview-ADB-Lineage-Solution-Accelerator/blob/release/2.2/docs/configuration.md#experimental-app-settings).  Is there ever a case where "://" would not be part of the prefix?

File: function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs [hunk 3]
```diff
@@ -228,11 +279,32 @@ private bool IsStartEventEnvironment(Event olEvent)
...
+                {
+                    if (outp.NameSpace.StartsWith(source)) return true;
+                }   
+            }
+            return false;
+        }
+
         private bool IsJoinEvent(Event olEvent)
         {
+            string[] special_cases = {"cosmos", "iceberg"};
```
Comment (wjohnson) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/OlMessageConsolodation.cs:304]: Does this get used in here?

File: function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/ValidateOlEvent.cs [hunk 4]
```diff
@@ -29,18 +29,39 @@ public ValidateOlEvent(ILoggerFactory loggerFactory)
...
+                }
+            }
+            return false;
+        }
+
         /// <summary>
         /// Performs initial validation of OpenLineage input
         /// The tested criteria include:
         /// 1. Events have both inputs and outputs
+        ///     a. Except for special cases covered in isDataSourceV2Event
```
Comment (wjohnson) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/OlProcessing/ValidateOlEvent.cs:56]: Could we add why this is necessary? Such as "OpenLineage does not emit a start event or complete event with all of the necessary information and so must be consolidated later on"?

**Comment by wjohnson** [comment]
A nitpicky thing :(

function-app/adb-to-purview/src/Function.Domain/Helpers/parser/DatabricksToPurviewParser.cs
function-app/adb-to-purview/src/Function.Domain/Helpers/parser/QnParser.cs
function-app/adb-to-purview/src/Function.Domain/Services/OlConsolodateEnrich.cs
function-app/adb-to-purview/src/Function.Domain/Services/OlToPurviewParsingService.cs
function-app/adb-to-purview/src/Functions/PurviewOut.cs

Do not include any actual code changes, just whitespace. Any chance you'd be willing to undo the whitespace changes just to keep the commit clean and focused on the files necessary to change?

**Comment by hmoazam** [comment]
Closing in favor of #145

**Pull request closed without merging**